docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+
+```typescript
+container.on("disconnected", () => {
+    // Prevent user edits to disable data loss.
+});
+
+container.on("connected", () => {
+    // Enable editing if disabled.
+});
+```
+
+Your code can reduce unnecessary network traffic by disconnecting when a user is idle. The following example assumes that there is a `user` object that emits *idle* and *active* events. 
+
+```typescript
+user.on("idle", () => {
+    // Disconnect the container when the user is idle.
+    container.disconnect();
+});
+
+user.on("active", () => {
+    // Connect the container when the user is active again.
+    container.connect();
+});
+```
+
+### Local Readiness state
+
+Local Readiness is a *client-relative state*: the container may have a different Local Readiness state on different clients. The following diagram shows the possible Local Readiness states and the events that cause a state transition. Details are below the diagram.
+
+<!-- TO MODIFY THIS DIAGRAM, SEE INSTRUCTIONS AT THE BOTTOM OF THIS FILE. -->
+![A state diagram of the two possible Local Readiness states](./images/LocalReadinessStates.png)
+
+#### Ready
+
+In this state, changes to the shared data objects can be made locally. They are shared with the Fluid service when, and only when, the container is in **published** and **connected** status. 
+
+A container is automatically in **ready** state as soon as it is created and it remains in this state unless it is disposed. 
+A disposed container transitions back to **ready** if its ID is passed to a call of `client.getContainer`.
+
+#### Disposed
+
+In scenarios where a container is no longer needed on the current client, you can dispose it with a call of `container.dispose()`. Disposing removes any server connections, so the Connection status becomes **disconnected**.
+There is a *disposed* event on the container object that you can handle to add custom clean up logic, such as removing registered events.
+The following shows the basic syntax:
+
+```typescript
+container.on("disposed", () => {
+    // Handle event cleanup to prevent memory leaks.
+});
+```
+
+It is a good practice to make sure all pending changes are saved prior to calling `dispose`.
+The following shows how to ensure that the disposal doesn't happen until the container is **saved**.
+
+```typescript
+function disposeWhenSafe {
+    if (container.isDirty) {
+        container.once("saved", () => {
+            container.dispose();
+        });
+    }
+}
+```
+
+After it is disposed, some properties of the `container` object can be read, but the data objects in the container can no longer be changed either locally of from data relayed by the Fluid service.
+
+Once a container is disposed, your code must call `client.getContainer` again if it needs to be reconnected.
+This also moves it back to the **ready** state.
```

Comment:
See a previous comment, I don't think we want to give the impression that a disposed/closed container can go back to the ready state. The underlying document needs to be loaded in a new container.